### PR TITLE
feat: add enforce_retention_duration param to vacuum method

### PR DESCRIPTION
**Pull request by houqp**
# Description

This maps to the reference implementation's
spark.databricks.delta.retentionDurationCheck.enabled config.

### CODE REVIEW

File: rust/src/delta.rs [hunk 1]
```diff
@@ -175,9 +175,14 @@ pub enum DeltaTableError {
     },
     /// Error returned when Vacuum retention period is below the safe threshold
     #[error(
-        "Invalid retention period, retention for Vacuum must be greater than 1 week (168 hours)"
+        "Invalid retention period, retention for Vacuum is configured to be greater than {} milliseconds, got {} milliseconds", .provided, .min
```
Comment (roeap) [inline rust/src/delta.rs:178]: should the order of `.provided` and `.min` be inverted?

**Review by roeap — COMMENTED** [review]
Looks good overall. Left a small comment for the error text.

File: rust/src/delta.rs (hunk 1 above)
Comment (houqp) [reply rust/src/delta.rs]: good catch!

**Review by roeap — APPROVED** [review]

File: rust/src/delta.rs [hunk 2]
```diff
@@ -175,9 +175,14 @@ pub enum DeltaTableError {
     },
     /// Error returned when Vacuum retention period is below the safe threshold
     #[error(
-        "Invalid retention period, retention for Vacuum must be greater than 1 week (168 hours)"
+        "Invalid retention period, retention for Vacuum is configured to be greater than {} milliseconds, got {} milliseconds", .min, .provided
```
Comment (wjones127) [inline rust/src/delta.rs:178]: Since the user provided hours, would it be possible to have the error message report back in hours instead of milliseconds?

Comment (houqp) [reply rust/src/delta.rs]: updated :+1:

**Review by wjones127 — APPROVED** [review]
Thanks, this looks great!

**Pull request merged**